src/xcode/ENA/ENA/Source/Services/__tests__/ExposureSubmissionServiceTests.swift

```diff
@@ -142,4 +142,92 @@ class ExposureSubmissionServiceTests: XCTestCase {
 
 		waitForExpectations(timeout: expectationsTimeout)
 	}
+
+	func testGetTestResult_success() {
+
+		// Initialize.
+
+		let keyRetrieval = MockDiagnosisKeysRetrieval(diagnosisKeysResult: (keys, nil))
+		let client = ClientMock()
+		let store = MockTestStore()
+		store.registrationToken = "dummyRegistrationToken"
+
+		let service = ENAExposureSubmissionService(diagnosiskeyRetrieval: keyRetrieval, client: client, store: store)
+		let expectation = self.expectation(description: "Expect to receive a result.")
+
+		// Execute test.
+
+		service.getTestResult { result in
+			expectation.fulfill()
+			switch result {
+			case .failure:
+				XCTFail("This test should always return a successful result.")
+			case .success(let testResult):
+				XCTAssertEqual(testResult, TestResult.positive)
+			}
+		}
+
+		waitForExpectations(timeout: .short)
+	}
+
+	func testGetTestResult_noRegistrationToken() {
+
+		// Initialize.
+		let expectation = self.expectation(description: "Expect to receive a result.")
+		let service = ENAExposureSubmissionService(
+			diagnosiskeyRetrieval: MockDiagnosisKeysRetrieval(diagnosisKeysResult: (keys, nil)),
+			client: ClientMock(submissionError: nil),
```

Comment:
sumissionError has default value `nil`, it's redundant here.

Reply:
I removed it here and from other places in the file where it was also redundant.